File: api/KT.Data/Repositories/EvolutionNoteRepository.cs
using System;
using KT.Data.Models;
using Common.Data.MongoDB;
using System.Threading.Tasks;
using System.Collections.Generic;
using MongoDB.Driver;

namespace KT.Data.Repositories
{
    public class EvolutionNoteRepository : MongoDBRepository<Guid, EvolutionNote>
    {
        public EvolutionNoteRepository(MongoDBContext context) : base(context, context.Collection<EvolutionNote>())
        {
        }

        public Task<List<EvolutionNote>> FindByClientId(Guid clientId)
        {
            var filter = _builder.Eq(e=> e.ClientId, clientId);
            return _collection.Find(filter)
                .ToListAsync();
        }
    }
}

File: client/api/repository/EvolutionNoteRepository.js
import {convertFromDatabase, createBaseRepository} from './MongoRepository'
import {ObjectId} from 'mongodb'

const EvolutionNoteRepository = createBaseRepository('EvolutionNote')

EvolutionNoteRepository.prototype.findByClientId = function (id) {
  const filters = {clientId: ObjectId(id)}
  return this.collection.find(filters)
    .toArray()
    .then(convertFromDatabase)
}

export default EvolutionNoteRepository
